File: dsp_modules/modules/wah/test/wah_test.c
#include <stdio.h>
#include "../../../test-tools/c/fopnchk/fopnchk.h"
#include <modules.h>

int main() {      
    printf("\n");
    printf("-------------\n");
    printf("-- TESTING --\n");
    printf("-------------\n");  

    float freq_low = 500;
    float freq_high = 1500;
    float freq_delta = 0.05;
    float Q = 0.1;
    
    int sampling_rate = 16000;
    int block_size = 128;

     // READ FILE
    FILE *f_in;
    f_in = fopen("../../../test-tools/signals/cleanguitar_sample.f32","rb");
    if (!f_in) {
        printf("Unable to open input file!");
        return -1;
    }  

    FILE *f_out = fopnchk("testvectors/c/output.f32","wb");

    if (!f_out) {
        printf("Unable to open output file!");
        return -1;
    }  
    
    t_wah_data wah_data;
    int init_code = wah_init(&wah_data, block_size,sampling_rate, freq_low, freq_high, freq_delta, Q);
    if (init_code!=0) printf("Init error code: %i\n", init_code);

    float inbuf[block_size];
    while (fread(inbuf, sizeof(float), block_size, f_in) == block_size)
    {
        wah_apply(&wah_data, inbuf);

        fwrite(&wah_data.output, sizeof(float), block_size, f_out);
    }

    fopnchk_close_all();    
    return 0;
}

File: dsp_modules/modules/wah/test/wah_test.m
clear all
BLOCKSIZE = 128;
FS = 16000;

addpath('../matlab');

% run and write testvector to file
if ~exist('testvectors/matlab','dir')
  mkdir('testvectors/matlab');
end
fo0 = fopen('testvectors/matlab/output.f32','wb');


[in, fs0] = audioread('../../../test-tools/signals/cleanguitar_sample.wav');

freq_low = 500;
freq_high = 1500;
freq_delta = 0.05;
Q = 0.1;
o = wah(BLOCKSIZE, FS, freq_low,freq_high,freq_delta,Q);

number_of_blocks = floor(length(in) / BLOCKSIZE);
output = [];

for i = 1:number_of_blocks   
    
    o.wah_apply(in(i*BLOCKSIZE - BLOCKSIZE + 1:i*BLOCKSIZE));
    output = [output; o.output_buffer];
   
end
% output = output./max(output);
% figure; plot(output);
% 
% maxyb = max(abs(output));
% output = output/maxyb;

audiowrite('wah.wav', output,FS)
fwrite(fo0, output, 'float32');
fclose('all');

